process/block/shardblock.go

```diff
@@ -1107,15 +1113,25 @@ func (sp *shardProcessor) requestMissingFinalityAttestingHeaders() uint32 {
 			sp.dataPool.HeadersNonces())
 
 		if err != nil {
-			requestedBlockHeaders++
-			go sp.onRequestHeaderHandlerByNonce(sharding.MetachainShardId, i)
+			missingFinalityAttestingHeaders++
+			wasHeaderRequested := sp.wasHeaderRequested(sharding.MetachainShardId, i)
+			if !wasHeaderRequested {
+				requestedBlockHeaders++
+				sp.hdrsForCurrBlock.requestedFinalityAttestingHdrs[sharding.MetachainShardId] = append(sp.hdrsForCurrBlock.requestedFinalityAttestingHdrs[sharding.MetachainShardId], i)
```

Comment:
you should protect with mutex

Reply:
The resource is already protected in upward methods from which this method is called: metaBlock: receivedShardHeader / requestShardHeaders
shardBlock: receivedMetaHeader / requestMetaHeaders